api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -1010,6 +1044,76 @@ static String notebookLockingEmailHash(String bucket, String email) {
     return ResponseEntity.ok(recentWorkspaceResponse);
   }
 
+  private WorkspaceResource createWorkspaceResourceFromWorkspace(DbWorkspace workspace) {
+    return new WorkspaceResource()
+        .workspaceFirecloudName(workspace.getFirecloudName())
+        .workspaceId(workspace.getWorkspaceId())
+        .workspaceNamespace(workspace.getWorkspaceNamespace())
+        .workspaceBillingStatus(workspace.getBillingStatus());
+  }
+
+  @Override
+  public ResponseEntity<WorkspaceResourceResponse> getDataPageItems(
+      String workspaceNamespace, String workspaceId) {
+    // This also enforces registered auth domain.
+    workspaceService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, workspaceId, WorkspaceAccessLevel.READER);
+
+    DbWorkspace workspace =
+        workspaceService.getRequiredWithCohorts(workspaceNamespace, workspaceId);
+    Set<DbCohort> cohorts = workspace.getCohorts();
+    WorkspaceResourceResponse workspaceResources = new WorkspaceResourceResponse();
+    workspaceResources.addAll(
+        cohorts.stream()
+            .map(cohortMapper::dbModelToClient)
+            .map(
+                cohort ->
+                    createWorkspaceResourceFromWorkspace(workspace)
+                        .cohort(cohort)
+                        .modifiedTime(cohort.getLastModifiedTime().toString()))
+            .collect(Collectors.toList()));
+    // TODO: Convert cohorts to resources
+    List<DbCohortReview> reviews =
+        cohortReviewService.getRequiredWithCohortReviews(workspaceNamespace, workspaceId);
+    workspaceResources.addAll(
+        reviews.stream()
+            .map(cohortReviewMapper::dbModelToClient)
+            .map(
+                cohortReview ->
+                    createWorkspaceResourceFromWorkspace(workspace)
+                        .modifiedTime(cohortReview.getLastModifiedTime().toString())
+                        .cohortReview(cohortReview))
+            .collect(Collectors.toList()));
+    // TODO convert cohort reviews to resources
+    List<DbConceptSet> conceptSets =
+        conceptSetService.findByWorkspaceId(workspace.getWorkspaceId());
+    workspaceResources.addAll(
+        conceptSets.stream()
+            .map(conceptSetMapper::dbModelToClient)
+            .map(
+                conceptSet ->
```

Comment:
This conversion can live in the ConceptSetMapper as well. We can have a `WorkspaceResource conceptSetToWorkspaceResource(DbConceptSet dbConceptSet, Workspace workspace)`. `WorkspaceMapper` has some examples of this.

We can do the same for the dataset, cohort, etc.